packages/api-fetch/src/index.js

```diff
@@ -130,14 +136,21 @@ function apiFetch( options ) {
 
 	const runMiddleware = ( index ) => ( nextOptions ) => {
 		const nextMiddleware = steps[ index ];
-		const next = runMiddleware( index + 1 );
-		return nextMiddleware( nextOptions, next );
+
+		if ( steps.length > ( index + 1 ) ) {
+			const next = runMiddleware( index + 1 );
+			return nextMiddleware( nextOptions, next );
+		}
+
+		// Next middleware is the last one, so no next arg is passed
```

Comment:
It's a subtle change in behavior that affects only internal usage; but I'm curious if there's any need or precedent in middleware patterns for omitting the `next` argument for the last in the stack vs. passing a noop (guaranteeing a middleware to receive `next` as a function).

Reply:
I did this mainly to simplify the expected arguments in the `toHaveBeenCalledWith` call of the unit test checking the behavior when using a custom fetch handler.

But you're right, I didn't remember the noop function, which is also easy to test and guarantees consistency since all middlewares will receive a next function. 

I'll apply this change in a bit.

Reply:
Done in 9bce879